<a href="https://colab.research.google.com/github/RogerCL24/ANN/blob/main/3_TensorFlow.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# Importamos TensorFlow

import tensorflow as tf

In [ ]:
mnist = tf.keras.datasets.mnist # mnist -> images of 0 to 9 digits

(x_train, y_train), (x_test, y_test) =  mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

11490434/11490434 [==============================] - 1s 0us/step


Build a `tf.keras.Sequential` model piling (**Arquitecture**).

In [ ]:
model = tf.keras.models.Sequential([
    tf.keras.layers.Flatten(input_shape=(28, 28)), # 1. Input layer: With 28x28 shape of the image, it gets flatten into a vector in order to be sended to the next layer
    tf.keras.layers.Dense(128, activation='relu'), # 2. Dense layer: Receive 128 neurons with ReLU function for the activation
    tf.keras.layers.Dropout(0.2),                  # 3. Droput layer: Prevents overfitting, clean the input randomly to not overload
    tf.keras.layers.Dense(10, activation='softmax')# 4. Last layer: Only can be 10 neurons (0 to 9 digit images)
])

# More parameters
model.compile(optimizer = 'adam',
              loss = 'sparse_categorical_crossentropy', # Performance assessment
              metrics = ['accuracy'])

Train & evaluate the defined model:

In [ ]:
model.fit(x_train, y_train, epochs=5) # repeat the training 5 times (epochs)

model.evaluate(x_test, y_test, verbose=2)

Epoch 1/5
1875/1875 [==============================] - 9s 5ms/step - loss: 0.2947 - accuracy: 0.9134
Epoch 2/5
1875/1875 [==============================] - 7s 4ms/step - loss: 0.1397 - accuracy: 0.9596
Epoch 3/5
1875/1875 [==============================] - 8s 4ms/step - loss: 0.1060 - accuracy: 0.9672
Epoch 4/5
1875/1875 [==============================] - 8s 4ms/step - loss: 0.0861 - accuracy: 0.9736
Epoch 5/5
1875/1875 [==============================] - 8s 4ms/step - loss: 0.0750 - accuracy: 0.9767
313/313 - 1s - loss: 0.0778 - accuracy: 0.9767 - 666ms/epoch - 2ms/step


[0.07784364372491837, 0.9767000079154968]

As you see the first epoch has a loss of 0.2947 and an accuracy of ``91.34%`` and the last epoch has a loss only of 0.0750 and an accuracy of almost ``98%``